From fc0680c9fd6d440629e2fba9a1394a4435ffd0fc Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Mon, 21 Nov 2011 15:42:17 +0800
Subject: [PATCH 0647/2463] ENGR00162701 mxc_vout: fix potential deadlock of
 fb_blank

add FBINFO_MISC_USEREVENT flag before calling fb_blank

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 drivers/media/video/mxc/output/mxc_vout.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/mxc/output/mxc_vout.c b/drivers/media/video/mxc/output/mxc_vout.c
index 661487e..c08f2fd 100644
--- a/drivers/media/video/mxc/output/mxc_vout.c
+++ b/drivers/media/video/mxc/output/mxc_vout.c
@@ -1351,7 +1351,9 @@ static int config_disp_output(struct mxc_vout_output *vout)
 		vout->disp_bufs[i] = fbi->fix.smem_start + i * display_buf_size;
 
 	console_lock();
+	fbi->flags |= FBINFO_MISC_USEREVENT;
 	ret = fb_blank(fbi, FB_BLANK_UNBLANK);
+	fbi->flags &= ~FBINFO_MISC_USEREVENT;
 	console_unlock();
 
 	return ret;
@@ -1363,7 +1365,9 @@ static void release_disp_output(struct mxc_vout_output *vout)
 	struct mxcfb_pos pos;
 
 	console_lock();
+	fbi->flags |= FBINFO_MISC_USEREVENT;
 	fb_blank(fbi, FB_BLANK_POWERDOWN);
+	fbi->flags &= ~FBINFO_MISC_USEREVENT;
 	console_unlock();
 
 	/* restore pos to 0,0 avoid fb pan display hang? */
@@ -1380,7 +1384,9 @@ static void release_disp_output(struct mxc_vout_output *vout)
 
 	if (get_ipu_channel(fbi) == MEM_BG_SYNC) {
 		console_lock();
+		fbi->flags |= FBINFO_MISC_USEREVENT;
 		fb_blank(fbi, FB_BLANK_UNBLANK);
+		fbi->flags &= ~FBINFO_MISC_USEREVENT;
 		console_unlock();
 	}
 }
-- 
1.7.7.4

